src/Rule/Trait/CountableLimitTrait.php

```diff
@@ -123,21 +123,25 @@ private function initCountableLimitProperties(
         }
 
         if (
-            ($this->min !== null && $this->min <= 0) ||
-            ($this->max !== null && $this->max <= 0) ||
-            ($this->exactly !== null && $this->exactly <= 0)
+            ($this->min !== null && $this->min < 0) ||
+            ($this->max !== null && $this->max < 0) ||
+            ($this->exactly !== null && $this->exactly < 0)
         ) {
-            throw new InvalidArgumentException('Only positive values are allowed.');
+            throw new InvalidArgumentException('Only positive or zero values are allowed.');
         }
 
+        $messageForUsingExactly = 'Use $exactly instead.';
+
         if ($this->min !== null && $this->max !== null) {
             if ($this->min > $this->max) {
                 throw new InvalidArgumentException('$min must be lower than $max.');
             }
 
             if ($this->min === $this->max) {
-                throw new InvalidArgumentException('Use $exactly instead.');
+                throw new InvalidArgumentException($messageForUsingExactly);
             }
+        } elseif ($this->min === null && $this->max === 0) {
+            throw new InvalidArgumentException($messageForUsingExactly);
```

Comment:
Max count equal to `0` is OK.  It's value can by dynamic:

`new Count(max: $max)` better than `new Count(max: $max> 0 ? $max : null, exactly: $max === 0 ? 0 : null)`.